observable.go

```diff
@@ -1601,6 +1610,17 @@ func (o *observable) TakeWhile(apply Predicate) Observable {
 	return newColdObservableFromFunction(f)
 }
 
+// Timestamp returns an Observable that emits items emitted by the source Observable
+// wrapped in a Timestamped struct containing the value and the timestamp when it was emitted
+func (o *observable) Timestamp() Observable {
+	return o.Map(func(item interface{}) interface{} {
+		return &Timestamped{
```

Comment:
I think I would prefer to return an observable of an interface instead of a struct.
This is a general direction I would say in RxGo. For example, before the v2 `Observable` was a struct. We switched to an interface which gives us more flexibility. I really feel like this is a best practice in general, exposing interfaces and hiding structures.

You can create a private struct implementing a public `Timestamped` interface offering two methods `value() interface` and `timestamp() time.Time`.

Reply:
Yes, that makes total sense. I wasn't sure how to represent the `value` and `timestamp` attributes on an interface, great Go lesson. Thanks :)